gremlin-go/driver/connection.go

```diff
@@ -55,8 +55,10 @@ type connectionSettings struct {
 func (connection *connection) errorCallback() {
 	connection.logHandler.log(Error, errorCallback)
 	connection.state = closedDueToError
-	err := connection.protocol.close()
-	if err != nil {
+
+	// This callback is called from within protocol.readLoop. Therefore,
```

Comment:
Can we modify protocol.readLoop?

Simply I would just like all `protocol.wg.Done()` to be removed, and instead added before the `for` as `defer(protocol.wg.Done()`.

Reply:
Sure thing. Done in c93661f. I did the same in `transporter.writeLoop`.